library/std/src/process.rs

```diff
@@ -646,9 +646,7 @@ impl Command {
         I: IntoIterator<Item = S>,
         S: AsRef<OsStr>,
     {
-        for arg in args {
-            self.arg(arg.as_ref());
-        }
+        self.inner.args(args.into_iter());
```

Comment:
This will cause us to codegen the full arg extension code for every different iterator passed in, which doesn't feel great to me (Command execution is high overhead, the extra code in binaries isn't worth it).

I think I'd be inclined to say that the small amount of allocations this saves isn't worth the extra code it's going to generate as a result into downstream binaries, so I'm inclined to not accept this PR. I think if we replaced this with taking something like `&mut dyn Iterator<Item = &OsStr>` I'd be okay r+ing that, though it's still not clear to me that additional code complexity is worthwhile here.

Did you start pursuing this for some particular reason? Benchmark or slow program?